### ISSUE TITLE
dio.download has two required parameters but official docs say 1.

### ISSUE THREAD

**Issue by theashggl** [cfug/dio#1511]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

I am able to see the number of parameters in the dio.download package as 2 one is the file path and one is the download path, but the official docs say that the method only has 1 required parameter. Attaching the screenshots of both official docs and method description from android studio.

![image](https://user-images.githubusercontent.com/43692609/175808918-54aa60dc-e940-4288-a4cc-d2e9c8ec5419.png)
![image](https://user-images.githubusercontent.com/43692609/175808946-a0a801df-ff9b-4c25-82cd-ef99101d0e6c.png)
